### DISCUSSION TITLE
Cannot find provider packages

### DISCUSSION THREAD

**Discussion by sxddhxrthx** [terra-farm/terraform-provider-virtualbox#128] [Support] [answered]
I am trying to create virtualbox vms using the the following code:

```tf
terraform {
  required_providers {
    virtualbox = {
      source = "terra-farm/virtualbox"
    }
  }
}

resource virtualbox_vm "k8s-node" {
  count = 2
  name = format("k8s-node-%02d", count.index + 1)
  image = "C:\\Users\\username\\Downloads\\CentOS-7-x86_64-Minimal-2009.iso"
  cpus = 2
  memory = "1536 mib"
  network_adapter = {
    type = "bridged"
    host_interface = "eth0"
  }
}
```

But when I run `terraform init`, all I get is the following error:

> Initializing the backend...
>
>Initializing provider plugins...
>- Finding latest version of terra-farm/virtualbox...
>  |
>│ Error: Failed to query available provider packages
>│
>│ Could not retrieve the list of available versions for provider terra-farm/virtualbox: no available releases match the given constraints

Not quite sure what am I doing wrong as I even tried copy-paste from official docs and changed the variable names but it didn't had impact on the result.

**Comment by VoyTechnology** [comment, marked as answer]
Hi, both the README and the [Terraform Registry](https://registry.terraform.io/providers/terra-farm/virtualbox/latest/docs) mention that you need to specify the version, which is also in your error message (`no available releases match the given constraints`).

This means that your `required_providers` block should look like this:

```tf
terraform {
  required_providers {
    virtualbox = {
      source = "terra-farm/virtualbox"
      version = "0.2.2-alpha.1"
    }
  }
}

provider "virtualbox" {
  # Configuration options
}
```

Let me know if it helps.

**Reply by sxddhxrthx** [reply]
I initially tried by specifying the version as "version = "0.2.1" which is mentioned in the docs but I had the same error. Then I thought maybe it would be a version specific error so I entirely removed the version attribute so that terraform may pick up the latest version but that didn't work either. 

Now when I use version = "0.2.2-alpha.1", it works fine. Thanks for your help!

**Reply by marsur** [reply]
> Hi, both the README and the [Terraform Registry](https://registry.terraform.io/providers/terra-farm/virtualbox/latest/docs) mention that you need to specify the version, which is also in your error message (`no available releases match the given constraints`).
> 
> This means that your `required_providers` block should look like this:
> 
> ```hcl
> terraform {
>   required_providers {
>     virtualbox = {
>       source = "terra-farm/virtualbox"
>       version = "0.2.2-alpha.1"
>     }
>   }
> }
> 
> provider "virtualbox" {
>   # Configuration options
> }
> ```
> 
> Let me know if it helps.

perfect. It helpful for me.
